cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/AbstractTimeSchedulerService.java

```diff
@@ -135,7 +136,7 @@ public static String getTriggerName(ProgramId program, SchedulableProgramType pr
   }
 
   public static String programIdFor(ProgramId program, SchedulableProgramType programType) {
-    return String.format("%s:%s:%s:%s:%s", program.getNamespace(), program.getApplication(), program.getVersion(),
-                         programType.name(), program.getProgram());
+    return String.format("%s:%s:%s:%s:%s", program.getNamespace(), program.getApplication(),
+                         ApplicationId.DEFAULT_VERSION, programType.name(), program.getProgram());
```

Comment:
I don't think we actually want to do this and the programId.toDefaultVersion() method. Let's change things to use the ProgramReference and ApplicationReference classes from the other PR. If it's not going to get merged soon, I suggest that we pull those classes out into a separate PR so that they can be used by multiple people.

Reply:
@albertshau Do you think we https://github.com/cdapio/cdap/pull/14671 is good enough to be approved/merged?

Reply:
Otherwise I can open a separate PR introducing `ProgramReference, ApplicationReference, ProgramRunReference`